src/tokens/Avatar/Avatar.cairo

```diff
@@ -140,58 +143,21 @@ func isApprovedForAll{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_chec
     return (isApproved,);
 }
 
-@view
-func getXp{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    tokenId: Uint256
-) -> (xp: Uint256) {
-    alloc_locals;
-    let account = (tokenId.high * 0x100000000000000000000000000000000) + tokenId.low;
-    let (xp) = IExperienceContract.balanceOf(
-        contract_address=EXPERIENCE_CONTRACT, 
-        account=account
-    );
-    return (xp=xp);
-}
-
 @view
 func tokenURI{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
     tokenId: Uint256
 ) -> (tokenURI_len: felt, tokenURI: felt*) {
     alloc_locals;
-    let (xp) = getXp(tokenId);
+    let (exp_addr) = Avatar_experience_contract.read();
+    let (xp) = IExperienceContract.balanceOf(
+        contract_address=exp_addr, 
+        account=tokenId
```

Comment:
I think we still need the conversion, account is felt type and tokenId is uint256